### ISSUE TITLE
Failing on icon

### ISSUE THREAD

**Issue by anderspitman** [linuxdeploy/linuxdeploy#75]
Hi there. I can't seem to get linuxdeploy to get past the icon stage. I have a icon.png file. If I include it in my AppDir (and add Icon=icon to my .desktop file, which is also in AppDir), I get:

`ERROR: Could not find icon executable for Icon entry: icon`

If I try and include the icon via -i, like this:

`./linuxdeploy-x86_64.AppImage --appdir AppDir/ --output appimage -i icon.png`

I get:

`ERROR: Icon icon.png has invalid x resolution: 400Failed to deploy icon: icon.png`

Any ideas? Is there any way to not have an icon? I'm packaging CLI tools and don't need them.

**Comment by TheAssassin** [comment]
You need to post more of the log, ideally the entire log you receive. That looks like a usage error, but to understand it, we need to see what messages precede your error message.

First random guess for first error message: you have a desktop file in your AppImage that references an icon file `icon` (by the way, you should pick `myapp` as icon name, that's cleaner). That's missing, as you correctly found. However, the icon file you provide has an invalid resolution of 400x<something>, but that's not allowed by the XDG icon standard. I don't recall the exact reference in the standard, but this is the list of valid resolutions:
https://github.com/linuxdeploy/linuxdeploy/blob/5c3a51fd487431022c8924d7a854a94ba8d64ad0/src/core/appdir.cpp#L516-L517 (in addition to "scalable", i.e., SVG icons).

We definitely need to improve that error message, and document this a bit better.

TL;DR: TODO: scale your icon file to a valid resolution, and then try again.

P.S.: There's a few resolutions missing actually in that list, it seems: 28, 36, 42, 384, 480. Will update the code asap.

**Comment by anderspitman** [comment]
Oooooh I thought "invalid x resolution" was referring to the xdg path resolution, ie it couldn't FIND the icon. I'll trying resizing and report back.

Also, that's not the actual name I'm using for the icon, I just used a generic one here to minimize the noise.

**Comment by anderspitman** [comment]
Ok @TheAssassin, resizing did the trick. Thanks!

**Comment by alaamroue** [comment]
For future reference:

If you get this error when the icon file is indeed in the directory.

Make sure the .desktop file is in **UNIX (LF)** Format .

This is especially relevant when working using WSL or when cross-compiling from Windows.

Try:
`dos2unix desktopfile.desktop`

**Comment by mariotaku** [comment]
@alaamroue This was exactly what I encountered, and fixing the line ending solved the issue. Thanks!
